connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/outbound/ConnectorJobHandler.java

```diff
@@ -179,12 +182,29 @@ public void handle(final JobClient client, final ActivatedJob job) {
               job.getCustomHeaders().get(Keywords.RESULT_VARIABLE_KEYWORD),
               job.getCustomHeaders().get(Keywords.RESULT_EXPRESSION_KEYWORD));
       result = new ConnectorResult.SuccessResult(response, responseVariables);
+    } catch (ConnectorRetryException ex) {
+      LOGGER.debug(
+          "ConnectorRetryException while processing job: {} for tenant: {}",
+          job.getKey(),
+          job.getTenantId(),
+          ex);
+      String errorCode = ex.getErrorCode();
+      result =
+          handleSDKException(
+              job,
+              ex,
+              Optional.ofNullable(ex.getRetries()).orElse(job.getRetries() - 1),
```

Comment:
It is the developer's responsibility to decrease the exception retries, we use this number as is.